x-pack/examples/ui_actions_enhanced_examples/public/dashboard_to_discover_drilldown/drilldown.tsx

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import { StartDependencies as Start } from '../plugin';
+import { reactToUiComponent } from '../../../../../src/plugins/kibana_react/public';
+import { StartServicesGetter } from '../../../../../src/plugins/kibana_utils/public';
+import { PlaceContext, ActionContext, Config, CollectConfigProps } from './types';
+import { CollectConfigContainer } from './collect_config_container';
+import { SAMPLE_DASHBOARD_TO_DISCOVER_DRILLDOWN } from './constants';
+import { UiActionsEnhancedDrilldownDefinition as Drilldown } from '../../../../plugins/advanced_ui_actions/public';
+import { txtGoToDiscover } from './i18n';
+
+const isOutputWithIndexPatterns = (
+  output: unknown
+): output is { indexPatterns: Array<{ id: string }> } => {
+  if (!output || typeof output !== 'object') return false;
+  return Array.isArray((output as any).indexPatterns);
+};
+
+export interface Params {
+  start: StartServicesGetter<Pick<Start, 'data'>>;
+}
+
+export class DashboardToDiscoverDrilldown
+  implements Drilldown<Config, PlaceContext, ActionContext> {
```

Comment:
> can just use an empty object?

Maybe it could be an empty object until it is used, but the correct type is `PlaceContext`.

Reply:
What do you mean by "correct type"?  This implementation is allowed to set the generic, as if it could set it to anything, but it actually has no control over this type.  I've posted in the slack channel about this, but I'll copy here.  I don't think `PlaceContext` should be a generic - it doesn't add any extra type safety, it actually will only hide type issues.

1) In hello world drilldown, I can put anything for `PlaceContext` and there are no errors:
<img width="675" alt="Screen Shot 2020-04-29 at 4 51 09 PM" src="https://user-images.githubusercontent.com/16563603/80725809-92911100-8ad1-11ea-8a20-9c13babd5534.png">

`context` passed into `CollectConfig` is still typed to a plain `object`.

2. I have to change the `CollectConfig` type to explicitly pass down the generic (which also seems to negate the point of having a generic at all, although it _did_ warn me if I tried to pass something explicit down that was the wrong type)
<img width="727" alt="Screen Shot 2020-04-29 at 4 54 44 PM" src="https://user-images.githubusercontent.com/16563603/80725896-b05e7600-8ad1-11ea-888b-994cdccde012.png">

^^ Now this looks type safe and I can access `context.foo.bar` as if it exists, but it doesn't.  This will throw an error at runtime, but pass all type checks.

Reply:
I've removed `PlaceContext` in this PR, so this is resolved now.

---

> What do you mean by "correct type"?

By correct I mean that this drilldown will receive a place context object of that type.

> This implementation is allowed to set the generic, as if it could set it to anything, but it actually has no control over this type.

This is a common practice, as I mentioned in one of the comments in this PR, it is the same for plugin definitions:

```ts
class MyPlugin extends Plugin<Anything, can, go, here> {

}
```

It is also the same for React components (note `123`):

```ts
const MyComponent: React.FC<123> = () => {
  return null;
};
```

and in many many other places.